include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -1064,4 +1069,80 @@ SplineBuilder<
             std::move(coefficients_derivs_xmax_out));
 }
 
+template <
+        class ExecSpace,
+        class MemorySpace,
+        class BSplines,
+        class InterpolationDDim,
+        ddc::BoundCond BcLower,
+        ddc::BoundCond BcUpper,
+        SplineSolver Solver,
+        class... IDimX>
+template <class KnotElement>
+void SplineBuilder<
+        ExecSpace,
+        MemorySpace,
+        BSplines,
+        InterpolationDDim,
+        BcLower,
+        BcUpper,
+        Solver,
+        IDimX...>::check_n_points_in_cell(int n_points_in_cell, KnotElement current_cell_end_idx)
+{
+    if (n_points_in_cell > BSplines::degree() + 1) {
+        KnotElement rmin_idx = ddc::discrete_space<BSplines>().break_point_domain().front();
+        int failed_cell = (current_cell_end_idx - rmin_idx).value();
+        throw std::runtime_error(
+                "The spline problem is overconstrained. There are "
+                + std::to_string(n_points_in_cell) + " points in the " + std::to_string(failed_cell)
+                + "-th cell.");
+    }
+}
+
+template <
+        class ExecSpace,
+        class MemorySpace,
+        class BSplines,
+        class InterpolationDDim,
+        ddc::BoundCond BcLower,
+        ddc::BoundCond BcUpper,
+        SplineSolver Solver,
+        class... IDimX>
+void SplineBuilder<
+        ExecSpace,
+        MemorySpace,
+        BSplines,
+        InterpolationDDim,
+        BcLower,
+        BcUpper,
+        Solver,
+        IDimX...>::check_valid_grid()
+{
+    const std::size_t n_interp_points = interpolation_domain().size();
+    std::size_t const expected_npoints
+            = ddc::discrete_space<BSplines>().nbasis() - s_nbc_xmin - s_nbc_xmax;
+    if (n_interp_points != expected_npoints) {
+        throw std::runtime_error(
+                "Incorrect number of points supplied to NonUniformInterpolationPoints. "
+                "(Received : "
+                + std::to_string(n_interp_points)
+                + ", expected : " + std::to_string(expected_npoints));
+    }
+    int n_points_in_cell = 0;
+    auto current_cell_end_idx = ddc::discrete_space<BSplines>().break_point_domain().front() + 1;
+    ddc::for_each(interpolation_domain(), [&](auto idx) {
+        ddc::Coordinate<continuous_dimension_type> point = ddc::coordinate(idx);
+        if (point == ddc::coordinate(current_cell_end_idx)) {
```

Comment:
No I don't think so. The less than and greater than cases are both covered. Equal to only needs treating separately because it is in 2 cells. If we are at knot+eps then there is no ambiguity so no special treatment is necessary

Reply:
I see, so if I understand correctly, in this case you assign the point to one cell ?

Reply:
Yes exactly. I can add a comment to explain this

Reply:
Comments added : [3e3d35c](https://github.com/CExA-project/ddc/pull/686/commits/3e3d35c0edb3d2e5adee167dd5583e09c348d6c1)